src/software/thunderscope/gl/layers/gl_tactic_layer.py

```diff
@@ -38,20 +36,17 @@ def __init__(self, name: str, buffer_size: int = 5) -> None:
     def refresh_graphics(self) -> None:
         """Update graphics in this layer"""
         self.cached_world = self.world_buffer.get(block=False)
-        play_info = self.play_info_buffer.get(block=False)
-        play_info_dict = MessageToDict(play_info)
+        play_info = self.play_info_buffer.get(block=False).GetOptions()
 
-        self.__update_tactic_name_graphics(
-            self.cached_world.friendly_team, play_info_dict
-        )
+        self.__update_tactic_name_graphics(self.cached_world.friendly_team, play_info)
 
-    def __update_tactic_name_graphics(self, team: Team, play_info_dict) -> None:
+    def __update_tactic_name_graphics(self, team: Team, play_info) -> None:
         """Update the GLGraphicsItems that display tactic data
 
         :param team: The team proto
-        :param play_info_dict: The dictionary containing play/tactic info
+        :param play_info: The dictionary containing play/tactic info
         """
-        tactic_assignments = play_info_dict["robotTacticAssignment"]
+        tactic_assignments = play_info.robotTacticAssignment()
```

Comment:
this doesn't quite work. The correct way access the robot tactic assignment field is: `play_info.robot_tactic_assignment()`.

This will return a `MessageMapContainer` which has this API: https://googleapis.dev/python/protobuf/3.17.0/google/protobuf/internal/containers.html#google.protobuf.internal.containers.MessageMap

The function call will match the field from the definition of the proto parameters: https://github.com/UBC-Thunderbots/Software/blob/10bded3efa3780478ae10a76840a5edf17abb508/src/proto/play_info_msg.proto#L18